protocol/x/clob/keeper/process_operations.go

```diff
@@ -532,7 +532,20 @@ func (k Keeper) PersistMatchLiquidationToState(
 	matchLiquidation *types.MatchPerpetualLiquidation,
 	ordersMap map[types.OrderId]types.Order,
 ) error {
-	takerOrder, err := k.MaybeGetLiquidationOrder(ctx, matchLiquidation.Liquidated)
+	// If the subaccount is not liquidatable, do nothing.
+	isLiquidatable, err := k.IsLiquidatable(ctx, matchLiquidation.Liquidated)
+	if err != nil {
+		return err
+	}
+	if !isLiquidatable {
+		return types.ErrSubaccountNotLiquidatable
+	}
```

Comment:
nit: this could be a function like `k.EnsureIsLiquidatable` that return an `error` since it's used twice. Up to you